saleor/dashboard/product/forms.py

```diff
@@ -326,3 +327,22 @@ def __init__(self, *args, **kwargs):
         product = kwargs.pop('product')
         super(UploadImageForm, self).__init__(*args, **kwargs)
         self.instance.product = product
+
+
+class ProductBulkUpdate(forms.Form):
+    """Performs one selected bulk action on all selected products."""
+    action = forms.ChoiceField(choices=ProductBulkAction.CHOICES)
+    products = forms.ModelMultipleChoiceField(queryset=Product.objects.all())
+
+    def save(self):
+        action = self.cleaned_data['action']
+        if action == ProductBulkAction.PUBLISH:
```

Comment:
PUBLISH to bool conversion (in ProductBulkAction?) might save you from this if-branching and redundant _(un)publish_products methods

Reply:
This branches seems fine, they are simple and readable. Functions `_publish_products` and `_unpublish_products` in this case may seem unnecessary. But in case one adds new bulk functions he will only have to add one more `IF` condition and one more handler function. I think it's a good design.